### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by adasescu**
Moved existing integrations tests to a new project. Modified build.proj to run integration tests.
Steps to start NAT server in go, on any operating system that have go installed:
1. from cmd, set gopath: 
	- C:\home_dir>mkdir gowp
	- C:\home_dir>set GOPATH=%CD%\gowp
	- C:\home_dir>set PATH=%CD%\gowp\bin;%PATH%
2. in a new cmd get gnatgs (NAT server in go) and start it:
	- go get github.com/apcera/gnatsd
	- gnatsd

### CODE REVIEW

File: src/CloudFoundry.Test.Integration/EnvVariables.cs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+using CloudFoundry.CloudController.V2.Client.Data;
+using CloudFoundry.UAA;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CloudFoundry.Test.Integration
```
Comment (viovanov) [inline src/CloudFoundry.Test.Integration/EnvVariables.cs]: These are not integration tests. They don't test the source code in the repo, but whatever may be deployed remotely.

File: src/CloudFoundry.Test.Integration/ApplicationTest.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CloudFoundry.Test.Integration
+{
+    [TestClass]
+    public class ApplicationTest
```
Comment (viovanov) [inline src/CloudFoundry.Test.Integration/ApplicationTest.cs]: These are not integration tests. They don't test the source code in the repo, but whatever may be deployed remotely.

File: src/CloudFoundry.Test.Integration/PushTest.cs [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CloudFoundry.Test.Integration
+{
+    [TestClass]
+    [DeploymentItem("Assets")]
+    public class PushTest
```
Comment (viovanov) [inline src/CloudFoundry.Test.Integration/PushTest.cs]: These are not integration tests. They don't test the source code in the repo, but whatever may be deployed remotely.

File: src/CloudFoundry.Test.Integration/ServiceInstanceTest.cs [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CloudFoundry.Test.Integration
+{
+    [TestClass]
```
Comment (viovanov) [inline src/CloudFoundry.Test.Integration/ServiceInstanceTest.cs]: These are not integration tests. They don't test the source code in the repo, but whatever may be deployed remotely.

File: src/CloudFoundry.Test.Integration/TestUtil.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+using System.Configuration;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CloudFoundry.Test.Integration
+{
```
Comment (viovanov) [inline src/CloudFoundry.Test.Integration/TestUtil.cs]: This utility class is for a client that tests a remote deployed system.

**Comment by viovanov** [comment]
The tests that push apps to a remote deployment are inproper. We must find a test strategy that tests the current sources, not a remote deployment (perhaps build, then attach the WinDEA to a running CloudFoundry using a unique stack).

Also, running integration tests should be exposed as a target in build.proj.

**Comment by mihaibuzgau** [comment]
verify this please

**Comment by mihaibuzgau** [comment]
verify this please

**Comment by mihaibuzgau** [comment]
verify this please

**Comment by mihaibuzgau** [comment]
verify this please

**Comment by mihaibuzgau** [comment]
verify this please

**Comment by mihaibuzgau** [comment]
verify this please

**Comment by mihaibuzgau** [comment]
verify this please

**Comment by mihaibuzgau** [comment]
verify this please

**Comment by mihaibuzgau** [comment]
verify this please

**Comment by mihaibuzgau** [comment]
verify this please

**Comment by mihaibuzgau** [comment]
:+1:

**Pull request merged**